src/main/java/games/strategy/engine/lobby/server/db/BannedMacController.java

```diff
@@ -13,51 +13,36 @@
 /**
  * Utilitiy class to create/read/delete banned macs (there is no update).
  */
-public class BannedMacController {
+public class BannedMacController extends TimedController {
   private static final Logger logger = Logger.getLogger(BannedMacController.class.getName());
 
   /**
-   * Ban the mac permanently.
-   */
-  public void addBannedMac(final String mac) {
-    addBannedMac(mac, null);
-  }
-
-  /**
    * Ban the given mac. If banTill is not null, the ban will expire when banTill is reached.
    *
    * <p>
    * If this mac is already banned, this call will update the ban_end.
    * </p>
    */
   public void addBannedMac(final String mac, final Instant banTill) {
-    if (isMacBanned(mac).getFirst()) {
-      removeBannedMac(mac);
-    }
-    Timestamp banTillTs = null;
-    if (banTill != null) {
-      banTillTs = new Timestamp(banTill.toEpochMilli());
-    }
-    try (final Connection con = Database.getPostgresConnection();
-        final PreparedStatement ps = con.prepareStatement("insert into banned_macs (mac, ban_till) values (?, ?)")) {
-      ps.setString(1, mac);
-      ps.setTimestamp(2, banTillTs);
-      ps.execute();
-      con.commit();
-    } catch (final SQLException sqle) {
-      if (sqle.getErrorCode() == 30000) {
-        // this is ok
-        // the mac is banned as expected
-        logger.info("Tried to create duplicate banned mac:" + mac + " error:" + sqle.getMessage());
-        return;
+    if (banTill == null || banTill.isAfter(now().plusSeconds(10L))) {
```

Comment:
I believe this check appears in all subclasses of `TimedController`.  Would it make sense to extract it to a new method in the base class?

What's the reasoning behind the magic 10 second delay?  Is it simply to avoid modifying the database for a "short-lived" ban/mute?  Do I need to know that when writing integration tests?

Reply:
It's thought to be some sort of a maximum execution time.
In case someone is banned 2 ms in the future, it will be the past until the statement is executed, and therefore the DB will raise an error.
The engine doesn't allow for such short-lived bans, so this is just to be sure we don't run into any issues.

Reply:
Extracting it to another method would probably not be very useful. 
The check is a one-liner, so we won't improve readability by a lot...

Reply:
> In case someone is banned 2 ms in the future, it will be the past until the statement is executed, and therefore the DB will raise an error.

Yeah, probably a minor issue, but those checks in the schema might end up being an over-constraint.  And the use of the Postgres `now()` function is another dependency on wall time.  Not sure how sensitive they'll be to system clock drift between different machines (e.g. when testing on Windows, my Postgres DB runs in a VM separate from where my client and lobby run).